src/fdcache.cpp

```diff
@@ -2142,6 +2150,9 @@ FdEntity* FdManager::Open(const char* path, headers_t* pmeta, ssize_t size, time
   if(0 != ent->Open(pmeta, size, time, no_fd_lock_wait)){
     return NULL;
```

Comment:
The caller cannot do this because the `FdEntity` is `NULL`.

Reply:
I think that ent is not NULL when calling ent->Open().
Open() can sometimes return an error if it can not be truncate and can not resize pagelist.
Therefore, you should call ent-> Close () before returning NULL with this function.

if(0 != ent->Open(pmeta, size, time, no_fd_lock_wait)){
  if(close){
    ent->Close();
  }
  return NULL;
}

Reply:
I will fix this part along with the defect of the reference count I found elsewhere.
So, I will close this review.

Reply:
Sorry I misunderstood this code -- you are correct.